lib/components/segmented-control/base.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {
+    KeyboardEvent,
+    ReactNode,
+    RefObject,
+    useEffect,
+    useState,
+} from "react";
+import { TRegularSizes } from "../../types";
+import { Segment } from "./segment";
+import "./segmented-control.scss";
+
+interface Option {
+    disabled?: boolean;
+    icon?: ReactNode;
+    label?: ReactNode;
+    selected?: boolean;
+}
+
+export interface SegmentedControlProps {
+    className?: string;
+    hasAnimation?: boolean;
+    options: Array<Option>;
+    onChange?: (selectedItemIndex: number) => void;
+    size?: TRegularSizes;
+}
+
+const KEY = {
```

Comment:
After both PR will be merged (yours and mine) we can move this object somewhere to utils in order to reuse, because most of the components will react with keyboard events